src/irreps/anirrep.jl

```diff
@@ -0,0 +1,304 @@
+"""
+struct ANIrrep{N} <: AbstractIrrep{Alternating{N}}
+    n::Int
+    function ANIrrep{N}(n::Int) where {N}
+        @assert 0 < N < 5 "ANIrrep is only provided for 0 < N < 5"
+        0 <= n <= _npartitions(N) - 1 ||
+        throw(DomainError(n, "ANIrrep only has irreps " * lazy"0 <= n <= $(_npartitions(N) - 1)"))
+        return new{N}(n)
+    end
+end
+
+_npartitions(N::Int) = (N == 3) ? 3 : (N == 4) ? 4 : 1
+
+Base.propertynames(x::ANIrrep) = (:n,)
+
+function Base.getproperty(a::ANIrrep{N}, sym::Symbol) where {N}
+    if sym === :n
+        return getfield(a, :n)
+    else
+        error("Unknown property $sym")
+    end
+end
+
+FusionStyle(::Type{ANIrrep{N}}) where {N} = N < 4 ? UniqueFusion() : GenericFusion()
+sectorscalartype(::Type{ANIrrep{N}}) where {N} = N < 4 ? Int64 : Float64
+Base.isreal(::Type{ANIrrep{N}}) where {N} = true
+
+unit(::Type{ANIrrep{N}}) where {N} = ANIrrep{N}(0)
+function dual(a::ANIrrep{N}) where {N}
+    N < 3 && return a
+    if N == 3
+        return ANIrrep{3}((3 - a.n) % 3)
+    else # N = 4
+        return a.n == 3 ? ANIrrep{4}(3) : ANIrrep{4}((3 - a.n) % 3)
+    end
+end
+
+Base.hash(a::ANIrrep, h::UInt) = hash(a.n, h)
+Base.convert(::Type{ANIrrep{N}}, n::Integer) where {N} = ANIrrep{N}(n)
+
+Base.getindex(::IrrepTable, ::Type{Alternating{N}}) where {N} = ANIrrep{N}
+
+const A4Irrep = ANIrrep{4}
+
+function Base.show(io::IO, a::ANIrrep)
+    if get(io, :typeinfo, nothing) !== typeof(a)
+        print(io, type_repr(typeof(a)))
+    end
+    print(io, a.n)
+    return nothing
+end
+
+# Sector iterator
+# ---------------
+Base.isless(a::ANIrrep{N}, b::ANIrrep{N}) where {N} = isless(a.n, b.n)
+Base.IteratorSize(::Type{SectorValues{ANIrrep{N}}}) where {N} = Base.HasLength()
+Base.length(::SectorValues{ANIrrep{N}}) where {N} = _npartitions(N)
+
+function Base.iterate(v::SectorValues{ANIrrep{N}}, i = 1) where {N}
+    return i > length(v) ? nothing : (v[i], i + 1)
+end
+
+@inline function Base.getindex(v::SectorValues{ANIrrep{N}}, i::Int) where {N}
+    L = length(v)
+    @boundscheck 1 <= i <= L || throw(BoundsError(v, i))
+    return ANIrrep{N}(i - 1)
+end
+
+findindex(::SectorValues{ANIrrep{N}}, a::ANIrrep{N}) where {N} = a.n + 1
+
+# Product iterator
+# ----------------
+
+const ANIrrepProdIterator{N} = SectorProductIterator{ANIrrep{N}}
+⊗(a::ANIrrep{N}, b::ANIrrep{N}) where {N} = SectorProductIterator((a <= b ? (a, b) : (b, a))...)
+
+Base.IteratorSize(::Type{ANIrrepProdIterator{N}}) where {N} = Base.HasLength()
+function Base.length(x::ANIrrepProdIterator{N}) where {N}
+    N < 4 && return 1 # abelian
+    a, b = x.a, x.b
+    if a == b
+        return a == 3 ? 5 : 1 # 3 x 3 = 1 + 1' + 1'' + 2*3
+    else
+        return 1
+    end
+end
+
+# TODO: any way to shorten this?
+function Base.iterate(p::ANIrrepProdIterator{N}, state::Int = 1) where {N}
+    a, b = p.a, p.b
+    if N < 4
+        return state > 1 ? nothing : (ANIrrep{N}((a.n + b.n) % _npartitions(N)), state + 1)
+    else
```

Comment:
Can we have an `elseif N==4` here, so as not to forget that this implementation is specific, and then perhaps also handle the `else` case with an error that this is not implemented (similarly as above).